normits_demand/models/forecasting/EDGE_growth/run.py

```diff
@@ -0,0 +1,315 @@
+# -*- coding: utf-8 -*-
+"""
+Created on: 17/05/2023
+Updated on:
+
+Original author: Ben Taylor
+Last update made by:
+Other updates made by:
+
+File purpose:
+
+"""
+# Built-Ins
+import pathlib
+from functools import partial
+import logging
+import os
+
+# Third Party
+import pandas as pd
+import numpy as np
+from tqdm import tqdm
+import caf.toolkit as ctk
+from caf.toolkit.concurrency import multiprocess
+
+# Local Imports
+# pylint: disable=import-error,wrong-import-position
+# Local imports here
+from normits_demand.models.forecasting.edge_growth import growth_matrices
+from normits_demand.models.forecasting.edge_growth import apply_growth
+from normits_demand.models.forecasting.edge_growth import utils
+from normits_demand.models.forecasting.edge_growth import loading
+from normits_demand.utils import timing
+from normits_demand.matrices import omx_file
+from normits_demand.utils import file_ops
+from normits_demand.models.forecasting import forecast_cnfg
+
+# pylint: enable=import-error,wrong-import-position
+
+# # # CONSTANTS # # #
+logging.basicConfig(level=logging.INFO)
+LOG = logging.getLogger(__name__)
+# # # CLASSES # # #
+
+# # # FUNCTIONS # # #
+def _tp_loop(
```

Comment:
I know this is only an internal function, but a more descriptive name would be helpful for review. It's not clear at the outset _why_ we are looping over time periods.